lib/parse.js

```diff
@@ -2417,11 +2417,16 @@ function parse($TEXT, options) {
             }
         }
 
-        var is_definition = is("keyword", "var") || is("keyword", "let") || is("keyword", "const");
+        var is_definition = is("keyword", "var")
+            || is("keyword", "let")
+            || is("keyword", "const")
+            || is("keyword", "function") && !is_default;
         if (is_definition) {
             exported_definition = statement();
+        } else if (is("keyword", "function")) {
+            exported_value = expr_atom(false);
         } else {
-            exported_value = expression();
+            exported_value = expression(false);
```

Comment:
`export 1,2;` is unnamed, so that's why it's an error.

`export default` can be unnamed but it correctly errors out at the comma:
```
$ echo 'export default (a,b);' | node_modules/.bin/acorn --module
{
  "type": "Program",
  "start": 0,
  "end": 22,
  "body": [
    {
      "type": "ExportDefaultDeclaration",
      "start": 0,
      "end": 21,
      "declaration": {
        "type": "SequenceExpression",
        "start": 16,
        "end": 19,
        "expressions": [
          {
            "type": "Identifier",
            "start": 16,
            "end": 17,
            "name": "a"
          },
          {
            "type": "Identifier",
            "start": 18,
            "end": 19,
            "name": "b"
          }
        ]
      }
    }
  ],
  "sourceType": "module"
}
```